src/app/data-explorer/selection.service.ts

```diff
@@ -35,14 +40,22 @@ export class SelectionService {
   private listSubject: Subject<DataElementSearchResult[]>;
 
   private STORAGE_KEY = 'elementSelection';
+  private SELECTION_TIMEOUT = 24 * 60 * 60 * 1000;
 
   constructor() {
+    let storedSelection: DataElementSearchResult[] | null = null;
     const storedJson = localStorage.getItem(this.STORAGE_KEY);
-    if (storedJson == null) {
+    if (storedJson != null) {
+      const stored = JSON.parse(storedJson) as StorageType;
+      if (stored.lastUpdated + this.SELECTION_TIMEOUT > Date.now()) {
```

Comment:
**Passed:** Selected items are being removed after 24 hours.
I made a small code change in my local copy to test this. The change I made was this:
```
      const testTime = Date.now() + this.SELECTION_TIMEOUT - 10000;
      if (stored.lastUpdated + this.SELECTION_TIMEOUT > testTime) {
```

I then ticked items and refreshed the page repeatedly. After 10 seconds the checks were removed. Therefore this is working as it should be.